csrc/host_ir/executor.h

```diff
@@ -47,20 +48,25 @@ class HostIrExecutor final : public OptInDispatch {
  public:
   HostIrExecutor(
       std::unique_ptr<HostIrContainer> container,
+      Communicator* communicator = nullptr,
```

Comment:
Question: there's also the approach of making the communicator a singleton that can be accessed globally. Do you prefer one way or the other? What would be the practical limitation if we chose singleton?

Reply:
using a singleton sounds good to me!

Reply:
I realize now we use one communicator with multiple process groups...

Reply:
> I realize now we use one communicator with multiple process groups...

yes, correct! So imo we are good for now